packages/react-slider/src/components/Slider/useSliderState.tsx

```diff
@@ -1,244 +1,60 @@
-
-      // eslint-disable-next-line deprecation/deprecation -- Should be remove an replaced with a useEvent hook.
-      disposables.current.push(on(target, 'pointermove', onPointerMove), on(target, 'pointerup', onPointerUp), () => {
-        target.releasePointerCapture?.(pointerId);
-      });
-
-      onPointerMove(ev);
-    },
-    [hideStepAnimation, onPointerDownCallback, onPointerMove, onPointerUp],
-  );
-
-  const onKeyDown = React.useCallback(
-    (ev: React.KeyboardEvent<HTMLDivElement>): void => {
-      hideStepAnimation();
-      onKeyDownCallback?.(ev);
-      const incomingValue = getKeydownValue(ev, currentValue, min, max, dir, keyboardStep);
-
-      if (currentValue !== incomingValue) {
-        updatePosition(incomingValue, ev);
-      }
-    },
-    [currentValue, dir, hideStepAnimation, keyboardStep, max, min, onKeyDownCallback, updatePosition],
-  );
-
-  const getTrackBorderRadius = () => {
-    if (origin !== undefined && origin !== (max || min)) {
-      if (vertical) {
-        return originPercent > valuePercent ? '99px 99px 0px 0px' : '0px 0px 99px 99px';
-      } else {
-        return (dir === 'rtl' ? valuePercent > originPercent : valuePercent < originPercent)
-          ? '99px 0px 0px 99px'
-          : '0px 99px 99px 0px';
-      }
-    }
-    return '99px';
-  };
-
-  useUnmount(() => {
-    disposables.current.forEach(dispose => dispose());
-    disposables.current = [];
-  });
-
-  const valuePercent = getPercent(renderedPosition !== undefined ? renderedPosition : currentValue, min, max);
-
+  const valuePercent = getPercent(currentValue, min, max);
   const originPercent = React.useMemo(() => {
     return origin !== undefined ? getPercent(origin, min, max) : 0;
   }, [max, min, origin]);
 
-  const markValues = React.useMemo((): number[] => getMarkValue(marks, min, max, step), [marks, max, min, step]);
+  const onChange: React.ChangeEventHandler<HTMLInputElement> = ev => {
+    const newValue = Number(ev.target.value);
+    setCurrentValue(newValue);
 
-  const markPercent = React.useMemo((): string[] => getMarkPercent(markValues), [markValues]);
+    const inputOnChange = state.input.onChange;
+    const propsOnChange = state.onChange;
 
-  const thumbWrapperStyles = {
-    transform: vertical
-      ? `translateY(${valuePercent}%)`
-      : `translateX(${dir === 'rtl' ? -valuePercent : valuePercent}%)`,
-    transition: stepAnimation ? `transform ease-in-out ${animationTime}` : 'none',
-    ...state.thumbWrapper.style,
+    if (inputOnChange && inputOnChange !== propsOnChange) {
+      inputOnChange(ev);
+    } else if (propsOnChange) {
+      propsOnChange(ev, { value: newValue });
+    }
   };
 
-  const trackStyles = {
-    [vertical ? 'top' : dir === 'rtl' ? 'right' : 'left']:
-      origin !== undefined ? `${Math.min(valuePercent, originPercent)}%` : 0,
-    [vertical ? 'height' : 'width']:
+  const rootVariables = {
+    '--slider-thumb-position': valuePercent + '%',
```

Comment:
Declare these css variable names as constants, so typos don't lead to bugs. Also, we should probably prefix them all with `--fui-Slider` to align with the className for the component.